mongodb/vibe/db/mongo/connection.d

```diff
@@ -181,6 +183,56 @@ final class MongoConnection {
 			throw new MongoDriverException(format("Failed to connect to MongoDB server at %s:%s.", m_settings.hosts[0].name, m_settings.hosts[0].port), __FILE__, __LINE__, e);
 		}
 
+		if (m_settings.nextgen) {
+			import os = std.system;
+			import compiler = std.compiler;
+			string platform = compiler.name ~ " "
+				~ compiler.version_major.to!string ~ "." ~ compiler.version_minor.to!string;
+			// TODO: add support for os.version
+
+			version (X86_64)
+				string arch = "x86_64 ";
+			else version (X86)
+				string arch = "x86 ";
+			else version (AArch64)
+				string arch = "aarch64 ";
+			else version (ARM_HardFloat)
+				string arch = "armhf ";
+			else version (ARM)
+				string arch = "arm ";
+			else version (PPC64)
+				string arch = "ppc64 ";
+			else version (PPC)
+				string arch = "ppc ";
+			else
+				string arch = "unknown ";
+
+			Bson handshake = Bson.emptyObject;
+			handshake["isMaster"] = Bson(1);
+			handshake["client"] = Bson([
+				"driver": Bson(["name": Bson("vibe.db.mongo"), "version": Bson("0.8.4")]),
+				"os": Bson(["type": Bson(os.os.to!string), "architecture": Bson(arch ~ os.endian.to!string)]),
+				"platform": Bson(platform)
+			]);
+
+			if (m_settings.appname.length) {
+				if (m_settings.appname.length > 128)
+					throw new MongoAuthException("The application name may not be larger than 128 bytes");
```

Comment:
enforce!MongoAuthException(m_settings.appname.length > 128, "The application name may not be larger than 128 bytes");